src/errors/checkXcmTxInputs.ts

```diff
@@ -436,64 +474,78 @@ const checkSystemAssets = async (
 			}
 		}
 	} else if (isLiquidTokenTransfer) {
-		await checkLiquidTokenValidity(api, systemParachainInfo, assetId);
+		await checkLiquidTokenValidity(api, registry, systemParachainInfo, assetId);
 	} else {
 		// check if assetId is a number
 		const parsedAssetIdAsNumber = Number.parseInt(assetId);
 		const invalidNumber = Number.isNaN(parsedAssetIdAsNumber);
 
 		if (!invalidNumber) {
-			// assetId is a valid number
-			// ensure the assetId exists as an asset on the system parachain
-			const assetSymbol: string | undefined =
-				systemParachainInfo.assetsInfo[assetId];
+			let assetSymbol: string | undefined = undefined;
+
+			// check the cache for the asset
+			// if not in cache, check the registry
+			if (registry.cacheLookupAsset(assetId)) {
+				assetSymbol = registry.cacheLookupAsset(assetId);
+			} else if (systemParachainInfo.assetsInfo[assetId]) {
+				assetSymbol = systemParachainInfo.assetsInfo[assetId];
+			}
 
 			if (assetSymbol === undefined) {
-				// if asset is not in registry, query the assets pallet to see if it has a value
-				const asset = await api.query.assets.asset(assetId);
+				// if asset is not in cache or registry, query the assets pallet to see if it has a value
+				const asset = await api.query.assets.asset(parsedAssetIdAsNumber);
 
 				// if asset is found in the assets pallet, return LocalTxType Assets
 				if (asset.isNone) {
 					throw new BaseError(
 						`(${xcmDirection}) integer assetId ${assetId} not found in ${specName}`,
 						BaseErrorsEnum.AssetNotFound
 					);
+				} else {
+					const assetSymbol = (
+						await api.query.assets.metadata(parsedAssetIdAsNumber)
+					).symbol
+						.toHuman()
+						?.toString();
+
+					const assetStr = assetSymbol as string;
+					// add the asset to the cache
+					registry.setAssetInCache(assetId, assetStr);
 				}
 			}
 		} else {
 			// not a valid number
 			// check if id is a valid token symbol of the system parachain chain
-			let isValidTokenSymbol = false;
+			if (assetId === '') {
```

Comment:
Is this suppose to error instead of return?

Reply:
since we allow `''` to represent the relay token,  here I've made it return early since it's valid. It will throw an error if all of the assetId checks fail/the asset isn't found or if multiple non unique matches of the given asset are found. Is there a different pattern you think works better here?

Reply:
Ahh perfect makes sense. Thanks for the explanation!